### ISSUE TITLE
JarHell Bootstrap Check can yield false positives

### ISSUE THREAD

**Issue by DJRickyB** [elastic/elasticsearch#75701]
For the JarHell Bootstrap Check, a user is seeing the following:

```[2021-07-22T11:47:41,971][ERROR][o.e.b.Bootstrap ] [node1] Exception
java.lang.IllegalStateException: failed to load plugin x-pack-core due to jar hell
at org.elasticsearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:524) ~[elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.plugins.PluginsService.loadBundles(PluginsService.java:469) ~[elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:163) ~[elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.node.Node.<init>(Node.java:313) ~[elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.node.Node.<init>(Node.java:257) ~[elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) [elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170) [elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:161) [elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) [elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125) [elasticsearch-cli-7.6.2.jar:7.6.2]
at org.elasticsearch.cli.Command.main(Command.java:90) [elasticsearch-cli-7.6.2.jar:7.6.2]
at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:126) [elasticsearch-7.6.2.jar:7.6.2]
at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) [elasticsearch-7.6.2.jar:7.6.2]
Caused by: java.util.zip.ZipException: error in opening zip file
at java.util.zip.ZipFile.open(Native Method) ~[?:1.8.0_201]
at java.util.zip.ZipFile.<init>(ZipFile.java:225) ~[?:1.8.0_201]
at java.util.zip.ZipFile.<init>(ZipFile.java:155) ~[?:1.8.0_201]
at java.util.jar.JarFile.<init>(JarFile.java:166) ~[?:1.8.0_201]
at java.util.jar.JarFile.<init>(JarFile.java:103) ~[?:1.8.0_201]
at org.elasticsearch.bootstrap.JarHell.checkJarHell(JarHell.java:183) ~[elasticsearch-core-7.6.2.jar:7.6.2]
at org.elasticsearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:510) ~[elasticsearch-7.6.2.jar:7.6.2]
```

Per the docstring on [checkJarHell](https://github.com/elastic/elasticsearch/blob/68817d7ca29c264b3ea3f766737d81e2ebb4028c/libs/core/src/main/java/org/elasticsearch/jdk/JarHell.java#L68):
```
     * @throws IllegalStateException if jar hell was found
```
But because `PluginsService.checkBundleJarHell()` [catches everything](https://github.com/elastic/elasticsearch/blob/68817d7ca29c264b3ea3f766737d81e2ebb4028c/server/src/main/java/org/elasticsearch/plugins/PluginsService.java#L570), the message obscures the fact that _the execution of the check itself_ fails

**Comment by rjernst** [comment]
I don't think this is exactly a false negative, as it's still a problem. We failed to read the jar file, so we could not check jar hell. The error message could certainly be better though, instead indicating that there was a general exception while trying to check the jar for consistency.

**Comment by DJRickyB** [comment]
Yep and also i meant false positive, whoops

**Comment by hrishikeshnikam2000** [comment]
Can I work on this issue? 

If yes: Since it is yielding false positives, should I make changes to `PluginsService.checkBundleJarHell()`  or should I make changes to the error message which checks the jar for consistency?

This is my first issue in this repo , sorry if any inconvenience that may have been caused.

**Comment by rjernst** [comment]
@hrishikeshnikam2000 It is marked as `help wanted`, so any help is appreciated. The change would need to be in `PluginsService.checkBundleJarHell()`. Currently there is a bare `catch (Exception)` there, but that picks up both the `IllegalStateException`s that jar hell checks throw, and other system level problems like missing or corrupt jar files, which in this case surface as a zip exception. Separating those two would allow a better error message, since in the general exception case it is not actually jar hell.

**Comment by BalasubramanyamEvani** [comment]
Hello @rjernst , could you please review my [PR](https://github.com/elastic/elasticsearch/pull/76217).

**Comment by rjernst** [comment]
This was fixed by #76217
